### DISCUSSION TITLE
Why aren't Oceans being generated after level 5

### DISCUSSION THREAD

**Discussion by mdoakes42** [onthegomap/planetiler#759] [General]
I generated a planet file for mbtiles using the OpenMaptiles profile.   I noticed that when I zoom in I lose the oceans after level 5.  (Possibly other large water bodies).      I then verified the "water" layer and see clearly after level 5 no mbtiles records are generated for oceans.     I used the --boundary-osm-only as a param. However, pretty sure I did this previously without this and had the same issue.    If you have any ideas please let me know.  Thanks

**Comment by msbarry** [comment]
What command did you run exactly? There's a few things that could be going on: 1) if you have a bounding box set then tiles intersecting your bounding box are included which sometimes makes it look like they disappear when you zoom in outside your bounding box or 2) ocean at z6 and above comes from the [water polygons source](https://osmdata.openstreetmap.de/data/water-polygons.html) - if that download failed or got corrupted then you could end up without any oceans above z5.

**Reply by mdoakes42** [reply]
This is what I ran 
java -Xmx64g -jar target/*with-deps.jar --osm-path=/data/planet.osm.pbf --transliterate=true --area=planet --boundary-osm-only --download --fetch-wikidata --water-polygons-url=https://github.com/onthegomap/planetiler/raw/main/planetiler-core/src/test/resources/water-polygons-split-3857.zip --natural-earth-url=https://github.com/onthegomap/planetiler/raw/main/planetiler-core/src/test/resources/natural_earth_vector.sqlite.zip --output=/share/mbtiles/osm_latest_12182023.mbtiles&

Shortly before this I ran a smaller area (either boston or monaco).  Is it possible that the water_polygons file was pulled for that smaller area and when I ran it for the planet file since the file already existed it didn't download it?

**Reply by mdoakes42** [reply]
I'm running that from the planetiler/planetiler-openmaptiles directory (using the osmmaptiler profile)

**Reply by mdoakes42** [reply]
-rw-r--r--. 1 root root    767047 Dec 15 15:27 water-polygons-split-3857.zip
-rw-r--r--. 1 root root   6502067 Dec 15 15:27 natural_earth_vector.sqlite.zip
-rw-r--r--. 1 root root  80961538 Dec 15 15:27 lake_centerline.shp.zip

Those are the size of the files that are in my directory

**Reply by msbarry** [reply]
Ah right those URLs are just for demo purposes when running over monaco so you don't need to download 1GB of data just to see it run.  You can exclude the `--*-url` parameters and it will download from the actual sources when you run with `--download` flag

**Reply by mdoakes42** [reply]
THank you, That was the issue.  I re-ran without those url arguments (last night) and just verified that I get the waters now at all zoom levels.  Thanks for your help.

**Comment by mdoakes42** [comment]
I think that might be what happend.   I re-downloaded the water-polygon file and it's a lot bigger

-rw-r--r--. 1 root root 846771837 Dec 19 04:48 water-polygons-split-3857.zip


Thanks for pointing that out.      Yeh, I didn't want to download any files becauase I already had the planet file and didn't want to spend the time downloading... But I guess that hurt me cause I didn't get the latest polygon.. Probably need to get the natural earth also and lake centerlines possibly.  I'll ck that out.  Thanks for pointing me in the right direction.  Let me know if you think this sounds right!    Appreciate all you did on this project also.  Works great!

**Comment by mdoakes42** [comment]
Actually not sure now.  When I use the standard URL ( as shown above in my command)  to get the polygons it looks like the 767K file  

But If I pull the file from here (https://osmdata.openstreetmap.de/data/water-polygons.html)  it's like 846 megs.  Do you know whats going on here?  Thanks
